ta4j-core/src/main/java/org/ta4j/core/BaseStrategy.java

```diff
@@ -181,7 +181,9 @@ public Strategy or(String name, Strategy strategy, int unstablePeriod) {
      * @param enter true if the strategy should enter, false otherwise
      */
     protected void traceShouldEnter(int index, boolean enter) {
-        log.trace(">>> {}#shouldEnter({}): {}", className, index, enter);
+        if ( log.isTraceEnabled() ) {
+            log.trace(">>> {}#shouldEnter({}): {}", getClass().getSimpleName(), index, enter);
```

Comment:
These were changed to use the string className instead of calling getSimpleName() on every single entry and exit.  IMO trace() should minimize or eliminate calls to other methods due to the potential frequency of execution.  See #90

Reply:
of course, i missed that change, updated the pr. log4j supports lambda logging, that eliminated the need for those ugly if blabla is enabled before the actual log statement, slf4j doent have that yet, so i think in this case it is still a valid performance improvement. it had a significant impact for me running parallel backtests, this methods used 15% CPU and now its arround 0.1% (trace disabled)

Reply:
You only see this 15% improvement with trace logging enabled, correct?  If you have trace logging enabled you really shouldn't be overly concerned with performance.  Just turn down the logging to run fast.  Turn up the logging to debug.  Putting if-enabled wrappers on every log call kind of defeats the purpose of having log levels in the first place.  Please let me know if I'm mistaken.  Also, maybe have a look at #103 and see if that could use some attention.  It hasn't been updated in almost a month.

Reply:
it was with trace disabled. its a very common pattern to surround those low level log statements because its faster to check than to build the strings every call. keep in mind that the message which is logged is build regardless if logging is enabled or no.